plugin/src/main/java/io/jenkins/plugins/casc/core/ProxyConfigurator.java

```diff
@@ -36,17 +37,18 @@ protected void configure(Mapping config, ProxyConfiguration instance, boolean dr
     protected ProxyConfiguration instance(Mapping mapping, ConfigurationContext context) throws ConfiguratorException {
         Set<String> keys = mapping.keySet();
         return new ProxyConfiguration(
-                mapping.getScalarValue("name"),
-                Integer.parseInt(mapping.getScalarValue("port")),
-                getOptionalParam(mapping, keys, "userName"),
-                getOptionalParam(mapping, keys, "password"),
-                getOptionalParam(mapping, keys, "noProxyHost"),
-                getOptionalParam(mapping, keys, "testUrl")
+                SecretSourceResolver.resolve(context, mapping.getScalarValue("name")),
```

Comment:
Should not be needed, now that I think about it. I believe using instance this way, is the wrong approach.
Best to use a describe with getter and setter. Perhaps best if we follow the approach noted here for custom configurators: https://github.com/jenkinsci/role-strategy-plugin/blob/c80586996e98a6984ffc754304a3d4659c7cb0ad/src/main/java/org/jenkinsci/plugins/rolestrategy/casc/RoleBasedAuthorizationStrategyConfigurator.java#L53-L70

Doing it that way you do not need to use getOptionalParam nor asking Secret Resolver :sweat:

Reply:
not sure that it will work that way, the ProxyConfiguration in core is missing getters / setters in places, and has optional fields in its constructor...

worth a go if it can be made to work

sidenote thats not documentation, and the configurators have no documentation really 😢

Reply:
but yeah seems very wrong to have to call secretresolver

Reply:
The reason we have to is that we are not calling `context.lookup` :cry:

Reply:
Please take a look at current variant. Is that what you meant?

Reply:
Looks alright apart from my latest comment